packages/augur-sdk/src/state/getter/Markets.ts

```diff
@@ -414,106 +442,153 @@ export class Markets {
     );
 
     let filteredKeyedMarketCreatedLogs = keyedMarketCreatedLogs;
-    if (params.search) {
-      const fullTextSearchResults = await db.fullTextMarketSearch(params.search);
 
-      const keyedFullTextMarketIds: any = fullTextSearchResults.reduce(
-        (previousValue: any, fullTextSearchResult: any) => {
-          previousValue[fullTextSearchResult.market] = fullTextSearchResult;
-          return previousValue;
-        },
-        []
-      );
+    if (params.search || params.categories) {
+      let keyedFullTextResults: any = {};
+      let searchResults: any = [];
+      if (params.search) {
+        searchResults = await db.fullTextMarketSearch(params.search, null);
+        keyedFullTextResults = _.keyBy(
+          searchResults,
+          (searchResult: MarketFields) =>  { return searchResult.market; }
+        );
+      }
+      let keyedCategoryResults: any = {};
+      if (params.categories) {
+        const extendedSearchOptions: ExtendedSearchOptions[] = [];
+        for (let i = 0; i < params.categories.length; i++) {
```

Comment:
prob need to filter out empty strings that might get passed in the category array.

Reply:
Yeah, I'll take another look at handling empty strings before merging.